test/e2e/cross_os/test_job_submissions.py

```diff
@@ -1437,3 +1437,160 @@ def check_percentage(complete_percentage) -> bool:
+                                {
+                                    "name": "attr.worker.os.family",
+                                    "allOf": [os.environ["OPERATING_SYSTEM"]],
+                                }
+                            ]
+                        },
+                        "script": {
+                            "actions": {
+                                "onRun": (
+                                    {"command": "{{ Task.File.runScript }}"}
+                                    if os.environ["OPERATING_SYSTEM"] == "linux"
+                                    else {
+                                        "command": "powershell",
+                                        "args": ["{{ Task.File.runScript }}"],
+                                    }
+                                ),
+                            },
+                            "embeddedFiles": [
+                                {
+                                    "name": "runScript",
+                                    "type": "TEXT",
+                                    "runnable": True,
+                                    "data": sleep_script,
+                                    **(
+                                        {"filename": "SleepScript.ps1"}
+                                        if os.environ["OPERATING_SYSTEM"] == "windows"
+                                        else {}
+                                    ),
+                                }
+                            ],
+                        },
+                    },
+                ],
+            },
+        )
+
+        @backoff.on_predicate(
+            wait_gen=backoff.constant,
+            max_time=120,
+            interval=2,
+        )
+        def is_job_started() -> bool:
+            job.refresh_job_info(client=deadline_client)
+            LOG.info(f"Waiting for job {job.id} to be created")
+            return job.lifecycle_status != "CREATE_IN_PROGRESS"
+
+        assert is_job_started()
+
+        @backoff.on_predicate(
+            wait_gen=backoff.constant,
+            max_time=180,
+            interval=4,
+        )
+        def get_session_action_id() -> Optional[str]:
+            sessions: list[dict[str, Any]] = deadline_client.list_sessions(
+                farmId=job.farm.id, queueId=job.queue.id, jobId=job.id
+            ).get("sessions")
+            for session in sessions:
+                session_actions: list[dict[str, Any]] = deadline_client.list_session_actions(
+                    farmId=job.farm.id,
+                    queueId=job.queue.id,
+                    jobId=job.id,
+                    sessionId=session["sessionId"],
+                ).get("sessionActions")
+                # There should be at most 1 sessionAction as there is only one task
+                assert len(session_actions) <= 1
+                if session_actions:
+                    return session_actions[0]["sessionActionId"]
```

Comment:
There should also only be one session right? Can we remove the `for` loop and replace with an assertion that only 1 session was created? Otherwise, we are just returning the first session action ID from one of potentially many sessions

Reply:
Okay, that totally makes sense I missed that. Will change